### PR TITLE
Resteasy Reactive: Inspect class tree for @BeanParam to find annotations

### LINKED ISSUE #21759: BeanParam extending another class fails in RESTClient
### Describe the bug

When a @BeanParam is used in a RESTClient and it extends another class it fails with the path param property not found.

```
Bad Request - Path parameter not provided client
```

If the class doe not extend another class or the properties are replicated in the extended class it will work fine.

[beanparam.zip](https://github.com/quarkusio/quarkus/files/7613800/beanparam.zip)

### Expected behavior

Classes can extend other classes and be used as @BeanParams.

### Actual behavior

When the class extends another class it fails on the parents "properties" with "property not provided"

### How to Reproduce?

Reproducer attached - running the tests demonstrate the issue.

There are two classes (`BaseParam` and `MethodParams`).  You will get the below exception:

```
java.lang.IllegalArgumentException: Path parameter not provided base
```

Prior to modifying the old reproducer (for RESTClient does not support BeanParams), and without extending another class, the test passed.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

JDK 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.5.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
The method `ClassInfo.annotations` only returns the annotations at the current class level. 
This PR makes the parser method to also find the inherited annotations.

Fixes https://github.com/quarkusio/quarkus/issues/21759

### CODE REVIEW

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java [hunk 1]
```diff
@@ -58,6 +69,40 @@ public String getId() {
...
+    public static class MyChildBeanParam extends MyBeanParam {
+        private final String base;
+
+        public MyChildBeanParam(String base, String id) {
+            super(id);
+            this.base = base;
+        }
+
+        @PathParam("base")
+        public String getBase() {
```
Comment (michalszynkiewicz) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java:81]: what happens if we override `getid()` here?

Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/beanparam/BeanPathParamTest.java:81]: anyway, this is a corner case that we can fix/specify later.

**Review by michalszynkiewicz — APPROVED** [review]
